1040.rb

```diff
@@ -0,0 +1,26 @@
+n1 = gets.to_f
+n2 = gets.to_f
+n3 = gets.to_f
+n4 = gets.to_f
+
+media = (2*n1 + 3*n2 + 4*n3 + 1*n4)/10
```

Comment:
Para melhorar a legibilidade, é interessante manter um espaço entre os operadores.
```suggestion
media = (2 * n1 + 3 * n2 + 4 * n3 + 1 * n4) / 10
```
Nesse caso, segundo a regra da matemática, ele já vai fazer a análise correta multiplicando antes de somar, mas se preferir deixar ainda mais legível, você pode usar o parenteses entre a aplicação da ponderação na nota, algo como:
```
media = ((2 * n1) + (3 * n2) + (4 * n3) + (1 * n4)) / 10
```